Sources/Core/iOS/ActivityIndicatorViewObserver.swift

```diff
@@ -0,0 +1,51 @@
+//
+//  ActivityIndicatorViewObserver.swift
+//  Operations
+//
+//  Created by Matthew Holden 16/04/2016.
+//
+//
+
+import UIKit
+
+public protocol ActivityIndicatorViewAnimationInterface {
+    func startAnimating()
+    func stopAnimating()
+}
+
+extension UIActivityIndicatorView: ActivityIndicatorViewAnimationInterface {}
+
+/**
+ An OperationObserverType that can update the state of an associated `UIActivityIndicatorView`
+ when the observed operation starts and finishes.
+ 
+ - note: Any type conforming to `ActivityIndicatorViewAnimationInterface` can be provided as 
+         the activity indicator.
+ */
+public class ActivityIndicatorViewObserver: OperationDidStartObserver, OperationDidFinishObserver {
+
+    private let activityIndicator: ActivityIndicatorViewAnimationInterface
+
+    /**
+     Initialize the observer with an `ActivityIndicatorViewAnimationInterface`-conforming type.
+     - parameter activityIndicator: the object to start and stop animating
+     - returns: an observer
+     - note: The activity indicator's `startAninating` and `stopAnimating` methods 
+             are guaranteed to execute on the main queue.
+     */
+    public init(activityIndicator: ActivityIndicatorViewAnimationInterface) {
```

Comment:
I had old-style Cocoa naming in my head when I wrote this (i.e. what would this initializer look like if it were originally written in Objective-C).  I'll update it.

This brings up something else I've been wondering, though.

If I read this line of code, in a vacuum:
```swift
// with or without the named argument:
let observer = ActivityIndicatorViewObserver(activityIndicator: myIndicator)
```

I would read it as "This gives me an object that _observes an Activity Indicator_"... not as something that will _update_ an activity indicator.

I get really in-depth about naming things, so this has been bugging me a bit.  

What do you think?